src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +45,104 @@
  * @version $Id$
  */
 public final class IndexesRequests implements SuperAgent {
+    /**
+     * Xpath expression for retrieving log entries from a talk's XML.
+     */
+    private static final String ARCHIVE_LOG = "//archive/log";
+
+    /**
+     * Name of the log index attribute.
+     */
+    private static final String INDEX = "index";
+
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int maxTalkIndex = this.getMaxTalkIndex(talks);
```

Comment:
@dpisarenko if you can't name all variables with simple nouns - something is wrong with the design and it needs refactoring. A method body should be easy to read. Simple variable names help us to read the method. When variable names are composite, they produce noise and reading is becoming more difficult and slow. So, when you need a composite name - refactor the method and make it shorter